plugin.py

```diff
@@ -99,3 +116,71 @@ def find_package_dependency_dirs(self, py_ver: Tuple[int, int] = (3, 3)) -> List
         dep_dirs.insert(0, os.path.join(self.package_storage(), "resources", "typings", "sublime_text"))
 
         return [path for path in dep_dirs if os.path.isdir(path)]
+
+    @classmethod
+    def resolve_python_path_from_venv(
+        cls, settings: DottedDict, workspace_folders: List[WorkspaceFolder]
+    ) -> Optional[str]:
+        """
+        Resolves the python binary path depending on environment variables and files in the workspace.
+
+        See https://github.com/fannheyward/coc-pyright/blob/d58a468b1d7479a1b56906e386f44b997181e307/src/configSettings.ts#L47.  # noqa: E501
+        """
+
+        def binary_from_python_path(path: str) -> Optional[str]:
+            if sublime.platform() == "windows":
+                binary_path = os.path.join(path, "Scripts", "python.exe")
+            else:
+                binary_path = os.path.join(path, "bin", "python")
+
+            return binary_path if os.path.isfile(binary_path) else None
+
+        python_path = settings.get("python.pythonPath")
+        if python_path:
+            return python_path
+
+        if not workspace_folders:
+            return None
+        workspace_folder = workspace_folders[0].path
+
+        # Config file, venv resolution command, post-processing
+        venv_config_files = [
+            ("Pipfile", ["pipenv", "--py"], None),
+            ("poetry.lock", ["poetry", "env", "info", "-p"], binary_from_python_path),
+            (".python-version", ["pyenv", "which", "python"], None),
+        ]  # type: List[Tuple[str, List[str], Optional[Callable[[str], Optional[str]]]]]
+
+        if sublime.platform() == "windows":
+            # do not create a window for the process
+            startupinfo = subprocess.STARTUPINFO()  # type: ignore
+            startupinfo.dwFlags |= subprocess.STARTF_USESHOWWINDOW  # type: ignore
+        else:
+            startupinfo = None  # type: ignore
+
+        for config_file, command, post_processing in venv_config_files:
+            full_config_file_path = os.path.join(workspace_folder, config_file)
+            if os.path.isfile(full_config_file_path):
+                try:
+                    python_path = (
+                        subprocess.check_output(command, cwd=workspace_folder, startupinfo=startupinfo)
+                        .decode("UTF-8")
```

Comment:
I suppose it could include `encoding='utf-8'` and `universal_newlines=True` instead of manual `decode`

Reply:
The `encoding` parameter did not exist in 3.3. But using just`universal_newlines` should work too.